### PR TITLE
MAINT: use OpenSearch 1.0.0-beta1 docker image for testing

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR

(1) Replaces ODFE docker image with OS staging docker image for tests including

- opensearch sink integration tests
- service map e2e tests

(2) Replaces class and variable names including ODFE with OpenSearch

Note: items remain to be replaced

(1) ODFE REST APIs: e.g. _opendistro/..
(2) ODFE image in examples
(3) ODFE or opendistro in docs
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public void testInstantiateSinkRawSpanDefault() throws IOException {
     assertFalse((boolean)mappings.get("date_detection"));
     sink.shutdown();
 
-    if (isODFE()) {
+    if (isOSBundle()) {
```
Comment (wrijeff) [inline data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java:99]: What is the purpose of `isODFE/isOSBundle`? From what I can tell it enables certain assertions and configures if a client should use http or https. A few follow-up questions:

1. Would the OpenSearchSink ever be tested on just OpenSearch (without bundled plugins)?
1. Why does bundle/no bundle change the test behavior?
1. Why does http/s depend on the bundle flag?

Comment (chenqi0805) [reply data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java:99]: > 1. Would the OpenSearchSink ever be tested on just OpenSearch (without bundled plugins)?
Yes. `./gradlew build` will run on OpenSearch core only.
> 2. Why does bundle/no bundle change the test behavior?
AFAICT, Bundle introduces
(1) security configuration (e.g. username/password) needed for security plugin
(2) Index management policy needed for ISM plugin 
> 3. Why does http/s depend on the bundle flag?
By default, bundle has security plugin enabled which requires https header.

Comment (wrijeff) [reply data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java:99]: Thanks for the info. Would suggest somehow changing the variable/method names (or even adding a comment) to better explain what the "isBundle" method is doing. Not blocking

**Review by wrijeff — APPROVED** [review]
👍

**Pull request merged**